### PR TITLE
fix(markdown): text and image overflow in markdown

### PR DESCRIPTION
**Pull request by vikram-raj**
Fix: https://github.com/openshiftio/openshift.io/issues/961

Before this PR:

screenshot from 2017-09-27 15-56-34

Now in this PR:

screenshot from 2017-09-27 15-58-41

### CODE REVIEW

File: src/app/markdown/markdown.component.less [hunk 1]
```diff
@@ -9,13 +9,15 @@
...
   background-color: @color-pf-black-200;
   .editor {
+    word-wrap: break-word;
     .editor-markdown {
       font-size: 1.071em;
         margin: 0;
     }
     .editor-preview {
-      h1 {
+      h1, h2, h3, h4, h5, h6 {
```
Comment (joshuawilson) [inline src/app/markdown/markdown.component.less]: I thought @mindreeper2420 said we were not supposed to use H1 or the others. @dlabrecq @SMahil do you remember?

Comment (sanbornsen) [reply src/app/markdown/markdown.component.less]: I think this is okay as it's a special case where it applies the styles in markdown-html element where you can't define a class.

Comment (SMahil) [reply src/app/markdown/markdown.component.less]: I don't think we should use heading tags here. If in any case we can't define classes we can put that tag into hierarchies and imply the styles

Comment (sanbornsen) [reply src/app/markdown/markdown.component.less]: What do you mean by hierarchy? It is in the maximum hierarchy position we can detect. Anything under editor-preview is server generated.

**Comment by SMahil** [comment]
I mean css hierarchies like outerclass> innerclass > inner element and so

**Comment by sanbornsen** [comment]
Yeah so **editor-preview** is the inner most class we have. Everything inside that is coming fromthe server.

File: src/app/markdown/markdown.component.less (hunk 1 above)
Comment (vikram-raj) [reply src/app/markdown/markdown.component.less]: @SMahil It is in hierarchal form  ```.editor-container > .editor.editor-preview > h1, h2, h3, h4, h5, h6```

**Comment by SMahil** [comment]
@vikramraj yes it is in css hierarchies but only thing i am saying instead of h1,h2 and so we can use some other tag

**Comment by sanbornsen** [comment]
Can you suggest edits in the code? I don't really understand what you mean. @SMahil

**Comment by vikram-raj** [comment]
@SMahil These tags are coming from the server. So, we can not replace these tags with other tags.

**Comment by mindreeper2420** [comment]
You should not need to add any line-heights that overwrite the master styles. Ideally, this would consume the line-heights from the typography rulesets.

What is most likely happening is that this is consuming the typography configuration from another area that is not setting the heading line-heights in a clean manner.

The ideal change would be to add a ’typography.less’ file in the assets directory of ngx-widgets (that matches Platform) which would then govern all elements of ngx-widgets.

**Comment by mindreeper2420** [comment]
@smahil @vikram-raj This file appears to be the main source of the issue: https://github.com/fabric8-ui/fabric8-planner/blob/master/src/assets/stylesheets/shared/_typography.less. 
If that is updated to match https://github.com/fabric8-ui/fabric8-ui/blob/master/src/assets/stylesheets/shared/_typography.less, the issue here should no longer occur.

**Comment by vikram-raj** [comment]
Thanks, @mindreeper2420, I will update the PR as you suggested.

**Comment by vikram-raj** [comment]
@mindreeper2420 I have some questions

- @SMahil said we are not supposed to use em. Then why fabric8-ui using em?
- Why are we defining our own style for header tags? Why don't  are we use default header tags style which is provided by patternfly?

So, In this PR I am just replacing the headers tags line height with default line height provided by patternfly.

**Comment by mindreeper2420** [comment]
@vikram-raj 
The custom typography was given to us by the visual design leads for PatternFly, in response to us bumping the base font to 14px. PatternFly uses 12px as the base, so we needed to updated line-heights and spacing accordingly. That is also why we are using `em` for the font sizes of the header tags. I'm not sure where the directive to not use `em` came from.

If you want the Markdown to match the rest of the UI, you should either have it match fabric8-ui, or change the Planner typography first. There is an issue for Planner typography overwriting Platform when you go from Platform to Planner and then back out to the rest of Platform.

**Comment by vikram-raj** [comment]
@mindreeper2420 Updated the typography file in the planner. PR https://github.com/fabric8-ui/fabric8-planner/pull/2204. And updated the PR accordingly.

**Comment by mindreeper2420** [comment]
@vikram-raj Awesome!

**Pull request merged**